src/views/Entries/ShareModal/index.tsx

```diff
@@ -72,6 +74,43 @@ function ShareModal(props: ComponentProps) {
         },
     );
 
+    const [
+        pendingReset,
+        ,
+        ,
+        triggerResetPublicUrl,
+    ] = useRequest<PublicUrl>(
+        {
+            url: `server://projects/${projectId}/public-viz/`,
+            method: 'POST',
+            body: resetBodyToSend,
+            onSuccess: (response) => {
+                if (onShareLinkChange) {
+                    onShareLinkChange(response?.publicUrl);
+                }
+            },
+            onFailure: (_, errorBody) =>
+                notifyOnFailure(_ts('entries', 'resetVizLink'))({ error: errorBody }),
+        },
+    );
+
+    const pending = pendingUrlVisibilityChange || pendingReset;
+
+    const copyToClipboard = () => {
+        document.addEventListener('copy', (e) => {
+            e?.clipboardData?.setData('text/plain', publicUrl ?? '');
+            e.preventDefault();
+        });
```

Comment:
Do we not have simpler method to copy to clipboard.
Also, we need to remove event listener